src/solver/impls/arkode/arkode.cxx

```diff
@@ -164,6 +165,15 @@ constexpr auto& ARKStepSetPreconditioner = ARKSpilsSetPreconditioner;
 constexpr auto& ARKStepSetUserData = ARKodeSetUserData;
 #endif
 
+#if SUNDIALS_VERSION_MAJOR >= 6
+#define SUNCTX_PLACEHOLDER , suncontext
```

Comment:
This is definitely clever, but it makes the source look _really_ strange. I think it might be nicer to just provide overloads/shims for previous versions (see earlier in this file for examples)